src/Microsoft.DotNet.ScenarioTests.FileBasedApps/FileBasedAppTests.cs

```diff
@@ -0,0 +1,359 @@
+using Microsoft.DotNet.ScenarioTests.Common;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.ScenarioTests.FileBasedApps;
+
+public class FileBasedAppTests : IClassFixture<ScenarioTestFixture>
+{
+    private readonly ScenarioTestFixture _scenarioTestInput;
+    private readonly FileBasedAppHelper _helper;
+
+    public FileBasedAppTests(ScenarioTestFixture testInput, ITestOutputHelper outputHelper)
+    {
+        if (string.IsNullOrEmpty(testInput.DotNetRoot))
+        {
+            throw new ArgumentException("sdk root must be set for file-based app tests");
+        }
+
+        _scenarioTestInput = testInput;
+        _helper = new FileBasedAppHelper(outputHelper, _scenarioTestInput.DotNetRoot, _scenarioTestInput.SdkVersion, _scenarioTestInput.BinlogDir);
+    }
+
+    [Fact]
+    [Trait("Category", "Offline")]
+    public void VerifySimpleFileBasedApp()
+    {
+        // Test simple HelloWorld.cs file execution
+        string testDir = Path.Combine(_scenarioTestInput.TestRoot, nameof(FileBasedAppTests), "SimpleApp");
+        Directory.CreateDirectory(testDir);
+
+        try
+        {
+            // Create a simple C# file
+            string csContent = @"using System;
+
+Console.WriteLine(""Hello World from file-based app!"");
+Console.WriteLine($""Current time: {DateTime.Now}"");";
+
+            string csFile = _helper.CreateCsFile("HelloWorld.cs", csContent, testDir);
+
+            // Execute the file using dotnet run
+            string output = _helper.ExecuteRunFile(csFile, testDir);
+
+            // Verify the output
+            Assert.Contains("Hello World from file-based app!", output);
+            Assert.Contains("Current time:", output);
+        }
+        finally
+        {
+            // Clean up
+            if (Directory.Exists(testDir))
+            {
+                Directory.Delete(testDir, true);
+            }
+        }
+    }
+
+    [Fact]
+    public void VerifyFileBasedAppWithPackageSearch()
+    {
+        // Test complex file-based app that searches for packages
+        string testDir = Path.Combine(_scenarioTestInput.TestRoot, nameof(FileBasedAppTests), "PackageSearchApp");
+        Directory.CreateDirectory(testDir);
+
+        try
+        {
+            // Create a complex C# file that searches for NuGet packages
+            string csContent = @"using System;
```

Comment:
The package search app should be loaded from the resources and not encoded directly into the test method.